src/org/zaproxy/zap/extension/tokengen/TokenGenerator.java

```diff
@@ -62,8 +77,11 @@ protected Void doInBackground() throws Exception {
 			HttpMessage msg = this.httpMessage.cloneRequest();
 
 			try {
-				msg.getRequestHeader().setHeader(HttpHeader.COOKIE, null);
+				// SB: Only remove the cookie that you want to analyse
+				// msg.getRequestHeader().setHeader(HttpHeader.COOKIE, null);
+				msg.getRequestHeader().setCookieParams(cookies);
 				this.getHttpSender().sendAndReceive(msg, true);
+				Thread.sleep (10); // Not to flood all the requests

```

Comment:
Removed.
I'll add a separate pull for this as it should be included, there only is a timing issue when you use this generator against an locally installed app.